ArcGISRuntimeSDKQt_CppSamples/Maps/ApplyScheduledMapUpdates/ApplyScheduledMapUpdates.cpp

```diff
@@ -0,0 +1,216 @@
+
+#include "Map.h"
+#include "MapQuickView.h"
+#include "MobileMapPackage.h"
+#include "OfflineMapSyncTask.h"
+#include "OfflineMapUpdatesInfo.h"
+#include "OfflineMapSyncParameters.h"
+#include "OfflineMapSyncJob.h"
+
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QtCore/qglobal.h>
+
+#ifdef Q_OS_IOS
+#include <QStandardPaths>
+#endif // Q_OS_IOS
+
+using namespace Esri::ArcGISRuntime;
+
+// helper method to get cross platform data path
+namespace
+{
+QString defaultDataPath()
+{
+  QString dataPath;
+
+#ifdef Q_OS_ANDROID
+  dataPath = "/sdcard";
+#elif defined Q_OS_IOS
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+#else
+  dataPath = QDir::homePath();
+#endif
+
+  return dataPath + "/ArcGIS/Runtime/Data/mmpk";
+}
+
+// sample MMPK location
+const QString sampleMmpk { "canyonlands" };
+
+// helper to copy directory from source to destination
+bool copyDir(const QString &srcPath, const QString &dstPath)
+{
+  const QDir parentDstDir(QFileInfo(dstPath).path());
+  if (!parentDstDir.mkdir(QFileInfo(dstPath).fileName()))
+    return false;
+
+  const QDir srcDir(srcPath);
+  for (const QFileInfo &info : srcDir.entryInfoList(QDir::Dirs | QDir::Files | QDir::NoDotAndDotDot))
+  {
+    const QString srcItemPath = srcPath + "/" + info.fileName();
+    const QString dstItemPath = dstPath + "/" + info.fileName();
+    if (info.isDir()) {
+      if (!copyDir(srcItemPath, dstItemPath)) {
+        return false;
+      }
+    } else if (info.isFile()) {
+      if (!QFile::copy(srcItemPath, dstItemPath)) {
+        return false;
+      }
+    } else {
+      qDebug() << "Unhandled item" << info.filePath() << "in copyDir";
+    }
+  }
+  return true;
+}
+}
+
+ApplyScheduledMapUpdates::ApplyScheduledMapUpdates(QObject* parent /* = nullptr */):
+  QObject(parent)
+{
+  // For the purposes of demonstrating the sample,
+  // create a temporary copy of the local offline map files,
+  // so that updating does not overwrite them permanently
+  copyDir(defaultDataPath() + "/" + sampleMmpk, m_TempDir.path() + "/" + sampleMmpk);
+}
+
+ApplyScheduledMapUpdates::~ApplyScheduledMapUpdates()
+{
+  if (m_mobileMapPackage)
+  {
+    m_mobileMapPackage->close();
+  }
+}
+
+void ApplyScheduledMapUpdates::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ApplyScheduledMapUpdates>("Esri.Samples", 1, 0, "ApplyScheduledMapUpdatesSample");
+}
+
+void ApplyScheduledMapUpdates::updateMap()
+{
+  if (!m_offlineSyncTask)
+    return;
+
+  m_offlineSyncTask->createDefaultOfflineMapSyncParameters();
+}
+
+MapQuickView* ApplyScheduledMapUpdates::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void ApplyScheduledMapUpdates::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+
+  // create MMPK
+  qDebug() << m_TempDir.path() + "/" + sampleMmpk;
```

Comment:
Did you intend to leave the code to print the path of the mmpk to qDebug?

Reply:
removed